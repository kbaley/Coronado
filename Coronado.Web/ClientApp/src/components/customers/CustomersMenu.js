import React, { Component } from 'react';
import { LinkContainer } from 'react-router-bootstrap';
import { NavItem, Glyphicon } from 'react-bootstrap';
import * as Mousetrap from 'mousetrap';
import { withRouter } from 'react-router-dom';

export class CustomersMenu extends Component {
  constructor(props) {
    super(props);
    this.goToCustomers = this.goToCustomers.bind(this);
    this.state = { isLoading: true };
  }

  componentDidMount() {
    Mousetrap.bind('g', this.goToInvoices);
  }

  goToCustomers() {
    this.props.history.push('/customers');
  }

  render() {
    return (
    <LinkContainer to={'/customers'}>
      <NavItem>
        <Glyphicon glyph='user' /> Customers
      </NavItem>
    </LinkContainer>
    );
  }
}

export default withRouter(CustomersMenu);