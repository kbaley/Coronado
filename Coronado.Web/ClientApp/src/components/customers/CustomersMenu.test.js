import { CustomersMenu } from "./CustomersMenu";
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router";

describe('CustomersMenu tests', () => {
  it('should render an CustomersMenu', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
        <CustomersMenu />
        </MemoryRouter>
      );
    expect(tree.toJSON()).toMatchSnapshot();
  });  

});