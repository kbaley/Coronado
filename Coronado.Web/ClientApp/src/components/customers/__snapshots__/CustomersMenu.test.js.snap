// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersMenu tests should render an CustomersMenu 1`] = `
<li
  className=""
  role="presentation"
>
  <a
    href="/customers"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="glyphicon glyphicon-user"
    />
     Customers
  </a>
</li>
`;
